CSUB-154 | new creditcoin integration tests

0xjeremyfrank opened the pull request:
Description of proposed changes:
Adding a new integration-tests project with clean typescript setup.

Practical tips for PR review & merge:

- [ ] All GitHub Actions report PASS
- [ ] Newly added code/functions have unit tests
  - [ ] Coverage tools report all newly added lines as covered
  - [ ] The positive scenario is exercised
  - [ ] Negative scenarios are exercised, e.g. assert on all possible errors
  - [ ] Assert on events triggered if applicable
  - [ ] Assert on changes made to storage if applicable
- [ ] Modified behavior/functions - try to make sure above test items are covered
- [ ] Integration tests are added if applicable/needed

Commit:
initial updated integration-tests project

Commit:
add add-ask-order and metrics tests

Commit:
creditcoin-js: add CreditcoinApi type

Commit:
integration-tests: add add-bid-order test, existing test cleanup

Commit:
integration-tests: add add-offer test

atodorov commented on integration-tests/src/test/add-ask-order.test.ts:
```diff
@@ -0,0 +1,59 @@
+import { Guid } from 'js-guid';
```
It seems like you've added these new files but the old ones are still present. 

B/c they show up as new files we can't see what's the difference with the existing ones. Maybe consider `git mv` to rename them instead ?

0xjeremyfrank replied:
The changes are drastic enough to setup that I just created a new "project", so not sure this will be possible. I don't want to get rid of the old before I'm done with the new so not sure `git mv` will work.

atodorov commented on integration-tests/src/test/add-ask-order.test.ts:
```diff
@@ -0,0 +1,59 @@
+import { Guid } from 'js-guid';
+import { KeyringPair } from '@polkadot/keyring/types';
```
From what I can tell CI is still using the old files to execute the integration tests. 

I don't see changes to `.github/workflows/ci.yml`. You will have to adjust which test files are used in the CI jobs before we can merge this PR.

0xjeremyfrank replied:
This is still a WIP. Once I have tests moved over I'll change CI.

atodorov reviewed (commented):
Another question:
- Where will type definitions be placed under the new directory structure? Is `creditcoin-js/src/interfaces/` the canonical location ? 

If yes will require updates to `javascript/prepare-devel-env.sh` and the CI config file.

Commit:
integration-tests: add add-offer tests

Commit:
integration-tests: add register-deal-order tests

Commit:
integration-tests: add register-funding-transfer tests

Commit:
integration-tests: add README.md

atodorov commented on integration-tests/src/test/common.ts:
```diff
@@ -2,8 +2,9 @@ import { Wallet } from 'ethers';
 import { BN } from '@polkadot/util';
 import { KeyringPair } from '@polkadot/keyring/types';
-import { Blockchain, LoanTerms } from 'creditcoin-js/model';
+import { Blockchain, LoanTerms, DealOrderId } from 'creditcoin-js/model';
```
I think prefixing with `../../../` to point to the actual file on disk will solve your issues. That's how some of the utils functions are imported in the existing test suite. They are not built as a JS library.

atodorov committed:
Enable the newly added tests in CI

copies & slightly modifies the existing prepare-devel-env.sh script
to make sure we wait properly for the Creditcoin node and also install
all of the necessary dependencies

Commit:
creditcoin-js: update project name in readme.md and package.json

Commit:
creditcoin-js: add type definition files to repo

atodorov committed:
tests: borrower should be Bob

to be consistent across the various scenarios

atodorov committed:
tests: Increase timeouts
